### DISCUSSION TITLE
Bold Tokens not being Detected

### DISCUSSION THREAD

**Discussion by rudra0713** [pymupdf/PyMuPDF#3943] [Q&A]
Hi, thanks again for the beautiful work on
[Sample_tables_rh.pdf](https://github.com/user-attachments/files/17352967/Sample_tables_rh.pdf)
 PDF documents.
I am facing an issue regarding the detection of bold tokens in the following document. Please check page 3 of the document, you will see tokens like Assets, Liabilities, etc in the first table in bold.  The document was created in Latex and the \textbf{} command was used to mark bold tokens. However, when I check the spans and the flags property all tokens are detected as not bold. I used the following code to determine such properties:
```
is_bold = bool(span['flags'] & 2) 
is_italic = bool(span['flags'] & 1)  
```
Please, let me know if this is an issue with the document or my code.
Thanks in advance.

**Comment by JorjMcKie** [comment]
One of the 2 fonts `CMBX10` is bold as the name suggests. The flag is incorrectly checked: `is_bold = bool(span['flags'] & 2**4)`. See documentation:
![image](https://github.com/user-attachments/assets/78704c9a-818c-4a73-bab0-02da6897c18d)

**Comment by rudra0713** [comment]
Thanks @JorjMcKie for your response and for fixing the bug in my code. I have a couple of questions:

1. What would you suggest as the most reliable way to detect bold tokens, using the flag bits or the font itself? Or, is it guaranteed that these two pieces of information will always be in sync?
2. As for the font name, can you kindly direct me to a page where the font names and their properties are explained? For example, CMBX10 is bold. I checked the following pages but could not find the information I was looking for:
https://pymupdf.readthedocs.io/en/latest/recipes-text.html
https://pymupdf.readthedocs.io/en/latest/app3.html#base-14-fonts
https://pymupdf.readthedocs.io/en/latest/font.html

**Reply by JorjMcKie** [reply]
1. The results between the span flags and the `Font` properties are always the same - because in both cases the same font binary is directly inquired.
2. As for the properties available for the `Font` class: indeed `.is_bold` etc. have not been mentioned yet. But you may check `help(pymupdf.Font)`. And then there is `Font.flags` - which **_is documented_** and shows this:
```python
pprint(pymupdf.Font("hebo").flags)
{'bold': 1,
 'cjk': 0,
 'cjk-lang': 0,
 'embed': 1,
 'fake-bold': 0,
 'fake-italic': 0,
 'invalid-bbox': 0,
 'italic': 0,
 'mono': 0,
 'never-embed': 0,
 'opentype': 0,
 'serif': 0,
 'stretch': 0,
 'substitute': 0}
```
